SYCL/ESIMD/api/functional/type_coverage.hpp

```diff
@@ -34,7 +34,7 @@ template <typename... T> struct named_type_pack {
       : names{std::forward<nameListT>(nameList)...} {}
 };
 
-enum class tested_types { all, fp, uint, sint };
+enum class tested_types { all, fp, fp_extra, uint, sint };
```

Comment:
Would you mind to add comment stating the general rules of differentiating the types between different groups?

Reply:
I've done it [here](https://github.com/intel/llvm-test-suite/pull/748/commits/928337781bcde4e2f1e8619464de3f9df60f7b54#diff-3bbc1ab1445c827282cc08d73f76b58cf553d9ced2c799ba46e6148583896d22R37-R42)

Reply:
So `all` doesn't include all other type groups, but only the C++ ones? Maybe we should rename `all` to the `core` (so we would have `core` and `extra` type groups) or to the `cpp`  (so we would have `cpp` and `non-cpp` type groups) then?

Nit: "all C++ signed/unsigned *integral* data types", because floating point type is signed

Reply:
> Maybe we should rename `all` to the `core`

I've done it [here](https://github.com/intel/llvm-test-suite/pull/748/commits/8d983fd269d836ba6c7611a1fca5eabc5f127235#diff-3bbc1ab1445c827282cc08d73f76b58cf553d9ced2c799ba46e6148583896d22R43)

> Nit: "all C++ signed/unsigned integral data types", because floating point type is signed

I've done it [here](https://github.com/intel/llvm-test-suite/pull/748/commits/8d983fd269d836ba6c7611a1fca5eabc5f127235#diff-3bbc1ab1445c827282cc08d73f76b58cf553d9ced2c799ba46e6148583896d22R41-R42)